services/accumulo-service/src/main/java/datawave/microservice/accumulo/lookup/LookupService.java

```diff
@@ -0,0 +1,400 @@
+import datawave.microservice.accumulo.lookup.config.LookupProperties;
+import datawave.microservice.audit.AuditClient;
+import datawave.microservice.authorization.user.ProxiedUserDetails;
+import datawave.security.authorization.DatawaveUser;
+import datawave.security.util.ScannerHelper;
+import datawave.webservice.query.exception.BadRequestQueryException;
+import datawave.webservice.query.exception.DatawaveErrorCode;
+import datawave.webservice.query.exception.NotFoundQueryException;
+import datawave.webservice.query.exception.QueryException;
+import datawave.webservice.response.LookupResponse;
+import datawave.webservice.response.objects.DefaultKey;
+import datawave.webservice.response.objects.Entry;
+import datawave.webservice.response.objects.KeyBase;
+import org.apache.accumulo.core.client.BatchScanner;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.io.Text;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import javax.annotation.Resource;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides Accumulo scan results in the form of {@link LookupResponse} for the given inputs (table, row, cf, cq, etc). Optionally supports auditing of lookup
+ * requests via autowired {@link AuditClient}
+ */
+@Service
+@ConditionalOnProperty(name = "accumulo.lookup.enabled", havingValue = "true")
+public class LookupService {
+    
+    public static final String ALLOWED_ENCODING = "[base64, none]";
+    
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    private enum Encoding {
+        none, base64;
+    }
+    
+    public interface Parameter {
+        String TABLE = "table";
+        String ROW = "row";
+        String ROW_ENCODING = "rowEnc";
+        String CF = "colFam";
+        String CF_ENCODING = "cfEnc";
+        String CQ = "colQual";
+        String CQ_ENCODING = "cqEnc";
+        String BEGIN_ENTRY = "beginEntry";
+        String USE_AUTHS = "useAuthorizations";
+        String END_ENTRY = "endEntry";
+    }
+    
+    @Resource(name = "auditLookupSecurityMarking")
```

Comment:
Shouldn't this be Autowired (with an `@Qualifier`) if you need a name. Same comments about finals an injecting in the constructor (at least IntelliJ tells me that's preferred).

Reply:
Nah, no big benefit I guess. [here](https://www.baeldung.com/spring-annotations-resource-inject-autowire) is a pretty decent overview. I had just been avoiding using the JSR-250 annotations, but I suppose if the code isn't going to get included in Wildfly there's no conflict.